pkg/sql/executor_ir.go

```diff
@@ -94,42 +94,47 @@ func SubmitWorkflow(sqlProgram string, modelDir string, session *pb.Session) *pi
 	return rd
 }
 
-func submitWorkflow(wr *pipe.Writer, sqlProgram string, modelDir string, session *pb.Session) (string, error) {
-	driverName, _, err := database.ParseURL(session.DbConnStr)
-	if err != nil {
-		return "", err
-	}
-	stmts, err := parser.Parse(driverName, sqlProgram)
-	if err != nil {
-		return "", err
-	}
-	// TODO(yancey1989): separate the IR generation to multiple steps:
-	// For example, a TRAIN statement:
-	// 		SELECT ... TO TRAIN ...
-	// the multiple ir generator steps pipeline can be:
-	// sql -> parsed result -> infer columns -> load train ir from saved model ..
+// ResolveSQLProgram accepts parse result from parser and returns a list of SQLFlowStmt
+func ResolveSQLProgram(sqlStmts []*parser.SQLFlowStmt, datasource, modelDir string, loadModel bool) ([]ir.SQLFlowStmt, error) {
```

Comment:
`ResolveSQLProgram` should not take `datasource, modelDir string, loadModel bool` arguments, since it only generates a workflow to run Couler

Reply:
Done.